.github/workflows/run-tests.yml

```diff
@@ -8,8 +8,44 @@ concurrency:
     group: ${{ github.workflow}}-${{ github.ref }}
     cancel-in-progress: true
 jobs:
+  check-formatting:
```

Comment:
Did you choose separate jobs for formatting & dependencies instead of putting them all in 1 job, since formatting does not need the code to be compiled?

Reply:
Same as before, for now I would maintain them separated and if everything works as we are expecting we can merge them in one single job

Reply:
Done, I grouped them together